applications/nrf_desktop/doc/hid_state.rst

```diff
@@ -3,5 +3,174 @@
 HID state module
 ################
 
-HID state module is responsible for aggregating data from user input sources
-and, depending on the state of the connection, transmitting it to the host.
+The |hid_state| is required for generating reports from input data.
+It is responsible for the following operations:
+
+    * Aggregating data from user input sources.
+    * Tracking state of the HID report subscriptions.
+    * Forming the HID reports.
+    * Transmitting the HID reports to the right subscriber.
+
+Module Events
+*************
+
+<TO DO>
+
+Configuration
+*************
+
+The |hid_state| module is enabled by selecting ``CONFIG_DESKTOP_HID_STATE_ENABLE``.
+This module is optional and turned off by default.
+
+Report expiration
+    With the ``CONFIG_DESKTOP_HID_REPORT_EXPIRATION`` configuration option, you can set the amount of time after which a key will be considered expired.
+    The higher the value, the longer the period after which the nRF Desktop application will recall pressed keys when the connection is established.
+
+Queue event size
+    With the ``CONFIG_DESKTOP_HID_EVENT_QUEUE_SIZE`` configuration option and similar options, you can set the number of elements on the queue where the keys are stored before the connection is established.
+    When a key is pressed before the connection is established, an element containing this key's usage is pushed onto the queue.
+    If there is no space in the queue, the oldest element is released.
+    For more information, see the `Implementation details`_ section.
+
+Implementation details
+**********************
+
+The |hid_state| areas of responsibility can be associated with the following modules:
+
+* Providing a routing mechanism between the input peripherals: :ref:`nrf_desktop_buttons`, :ref:`nrf_desktop_wheel`, and :ref:`nrf_desktop_motion`.
+* Sending out HID reports using :ref:`nrf_desktop_hids` and :ref:`nrf_desktop_usb_state`.
+
+For the routing mechanism to work, the module performs the following activities:
+
+    * `Linking input data with the right HID report`_
+    * `Storing input data before the connection`_
+    * `Tracking state of transports`_
+    * `Tracking state of HID report notifications`_
+    * `Forming HID reports`_
+
+These activities are described in the following sections.
+
+Linking input data with the right HID report
+============================================
+
+Out of all available input data types, the following types are collected by the |hid_state|:
+
+    * `Relative value data`_ (axes)
+    * `Absolute value data`_ (buttons)
+
+Both types are stored in the :c:type:`struct report_data` structure.
+
+Relative value data
+-------------------
+
+This type of input data is related to the pointer coordinates and the wheel rotation.
+Both ``motion_event`` and ``wheel_event`` are sources of this type of data.
+
+To indicate a change to this input data, overwrite the value that is already stored.
```

Comment:
this text is simplification of what code does. keys are absolute and axes are relative. we could keep the value 1 or 0 for the key but we need to track if there is more then one key with the same usage that is being pressed. I don't want to state everything here as it will bloat the documentation with details that anybody can get by having a quick look into the code.

Reply:
Ok so let's leave it as it is.